lib/ansible/plugins/terminal/__init__.py

```diff
@@ -48,7 +48,8 @@ class TerminalBase(with_metaclass(ABCMeta, object)):
 
     #: compiled bytes regular expressions to remove ANSI codes
     ansi_re = [
-        re.compile(br'(\x1b\[\?1h\x1b=)'),
+        # see ECMA-48 Section 5.4 (Control Sequences)
+        re.compile(br'((?:\x9b|\x1b\x5b)[\x30-\x3f]*[\x20-\x2f]*[\x40-\x7e])'),
```

Comment:
The regex is already present as part of aruba terminal plugin https://github.com/ansible-collections/community.network/blob/main/plugins/terminal/aruba.py#L35
Other platforms have not reported this issue any specific reason it is added in the general regex list. Also apart from aruba have you tested it with any other network os?

Reply:
Well, 3 1/2 years ago, when I created this pull request, that regex was not present on Aruba. The reason it belongs in this file is that it's not vendor-specific. ANSI control sequences are special characters used in an SSH session to tell your terminal emulator things like where to move the cursor, when to clear the screen, how to format things, etc. When receiving output programmatically, these control sequences ought to be completely ignored. The line deleted in this change filters out a _**subset**_ of these control sequences, which is apparently good enough for most vendors. Apparently someone copied my change and managed to get it merged for Aruba only, but there's no reason some new product or a new vendor might not also use ANSI control sequences other than those caught by the existing filter, in which case some poor intern will have to dig through a massive library to figure out what to change to get his project to work. The regex proposed in this change filters out _**specifically**_ every valid ANSI control sequence, which is why the comment directs the reader to consult the official specification. It couldn't possibly break any other platform, unless someone is trying to glean useful data from meaningless cursor commands. Now, I haven't used Ansible for years, so in that regard I don't care one bit whether you merge this or not, but it's a good change and I stand by it.